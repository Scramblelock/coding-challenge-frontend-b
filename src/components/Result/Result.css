.result-wrapper {
  background: white;
  border-radius: 20px;
  margin: 15px 120px;
  padding: 20px;
  max-width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  align-items: center;
  /* justify-content: space-around; */
}

.departure-result {
  padding-left: 30%;
  text-align: left;
}

.arrival-result {
  text-align: left;
  padding-left: 40%;
}

.location {
  font-size: 12px;
  color: rgb(118, 118, 119);
}

.price-result {
  padding-bottom: 20px;
}
.buy-button {
  background: #82cab7;
  border: none;
  border-radius: 12px;
  padding: 0px 30px;
  outline: none;
  color: white;
}

.buy-button:hover {
  cursor: pointer;
}

@media only screen and (max-width: 768px) {
  .result-wrapper {
    margin: 15px 40px;
  }

  .departure-result,
  .arrival-result {
    text-align: center;
    padding-left: 0;
  }
}

@media only screen and (max-width: 414px) {
  .result-wrapper {
    display: flex;
    flex-direction: column;
    margin: 15px 20px;
  }
}
