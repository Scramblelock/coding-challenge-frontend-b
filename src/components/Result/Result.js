import React from "react";
import { useTranslation } from "react-i18next";
import "./Result.css";

const Result = ({
  departureTime,
  arrivalTime,
  tripPrice,
  destinationLocationName,
  originLocationName
}) => {
  const { t } = useTranslation();
  return (
    <div className='result-wrapper'>
      <div className='departure-result'>
        <h2>{t("Departs")}</h2>
        <h4>New York</h4>
        <p>{departureTime}</p>
        <h5 className='location'>{originLocationName}</h5>
      </div>
      <div className='arrival-result'>
        <h2>{t("Arrives")}</h2>
        <h4>Montreal</h4>
        <p>{arrivalTime}</p>
        <h5 className='location'>{destinationLocationName}</h5>
      </div>
      <div className='price-result'>
        <h2>${tripPrice} CAD</h2>
        <button className='buy-button'>
          <h2>{t("Buy Now")}</h2>
        </button>
      </div>
    </div>
  );
};

export default Result;
