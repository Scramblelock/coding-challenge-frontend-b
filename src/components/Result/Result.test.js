import { shallow } from "enzyme";
import React from "react";
import Result from "./Result";

it("expect to render Result component", () => {
  expect(shallow(<Result />).debug()).toMatchSnapshot();
});
