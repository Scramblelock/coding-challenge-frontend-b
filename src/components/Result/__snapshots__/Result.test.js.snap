// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Result component 1`] = `
"<div className=\\"result-wrapper\\">
  <div className=\\"departure-result\\">
    <h2>
      Departs
    </h2>
    <h4>
      New York
    </h4>
    <p />
    <h5 className=\\"location\\" />
  </div>
  <div className=\\"arrival-result\\">
    <h2>
      Arrives
    </h2>
    <h4>
      Montreal
    </h4>
    <p />
    <h5 className=\\"location\\" />
  </div>
  <div className=\\"price-result\\">
    <h2>
      $
       CAD
    </h2>
    <button className=\\"buy-button\\">
      <h2>
        Buy Now
      </h2>
    </button>
  </div>
</div>"
`;
